### ISSUE TITLE
allyground_attackrange_gl4 reporting zombies

### ISSUE THREAD

**Issue by saurtron** [beyond-all-reason/Beyond-All-Reason#4031]
### Description

This is because engine isn't actually removing from selection handler when units get killed. Can be solved by https://github.com/beyond-all-reason/spring/pull/1824 if that gets accepted, otherwise we can work around, but there might be an unknown number of widgets/gadgets with the same problem.

### Expected Behaviour

Shouldn't be reporting zombies, specially in xmas XD

### Actual Behaviour

Reports zombies

### Reproduction steps

Can be reproduced using the test and dbg_test_runner "fix" at this branch: https://github.com/saurtron/Beyond-All-Reason/tree/test-selection-destroy check instructions at the above engine issue.

### Other

```
[t=00:47:39.115268][f=0021436] TraceFullEcho:[allyground_attackrange_gl4	]
1: popElementInstance  @LuaUI/Widgets/Include/instancevbotable.lua:491 Locals:(iT={myName:allyground_attackrange_gl4, primitiveType:4, VAO:<userdata>, zombies:<table>, makeVAOandAttach:<function>, }[#28]; instanceID=11100811; noUpload=nil; oldElementIndex=1; lastElementIndex=7; lastElementInstanceID=1125756; iTStep=24; endOffset=144; oldOffset=0; instanceData={1:3944.63745, 2:316.712006, 3:1124.28015, 4:140, 5:1, }[#1200])
2: RemoveSelectedUnit  @LuaUI/Widgets/gui_attackrange_gl4.lua:585 Locals:(unitID=811; mouseover=true; collections={811:<table>, }[#1]; removedRings=0; collection={posx:3275.30249, posy:315.015625, allied:true, unitDefID:99, vaokeys:<table>, }[#6]; (for generator)=<function>; (for state)={11100811:allyground, }[#1]; (for control)=11100811; instanceKey=11100811; vaoKey=allyground)
3: ??
4: pcall  @=[C]:-1 Locals:()
5: Update  @LuaUI/barwidgets.lua:711 Locals:(w={DrawWorldPreUnit:<function>, uploadAllElements:<function>, LUAUI_DIRNAME:LuaUI/, VFS:<table>, tostring:<function>, }[#103])
6: func  @LuaUI/barwidgets.lua:1276 Locals:(self=??; deltaTime=0.0053014; (for generator)=<function>; (for state)={1:<table>, 2:<table>, 3:<table>, 4:<table>, 5:<table>, }[#76]; (for control)=16; _=16; w={DrawWorldPreUnit:<function>, uploadAllElements:<function>, LUAUI_DIRNAME:LuaUI/, VFS:<table>, tostring:<function>, }[#103])
7: Update  @LuaUI/barwidgets.lua:852 Locals:()
8: ??
```

**Comment by saurtron** [comment]
> Isn't it expected for there to be zombies until the next frame? GL4 batches updates to GameFramePost.

I don't think so, otherwise why the warnings? The proposed fix solves this too, note in this case it's just a warning, but in other similar situations it can be worse. Actually in this case it's not so good because I think that requires the vbo code to be doing a lot of extra checks to ensure consistency.

**Comment by sprunk** [comment]
> I think that requires the vbo code to be doing a lot of extra checks to ensure consistency.

Make the VBO use RenderUnitDestroyed, which is the one after which the unit is definitely gone. There isn't supposed to be any inconsistency at any point before this (incl. after UnitDestroyed).

**Comment by saurtron** [comment]
> Make the VBO use RenderUnitDestroyed, which is the one after which the unit is definitely gone. There isn't supposed to be any inconsistency at any point before this (incl. after UnitDestroyed).

but that's too late, all those widgets don't really want the destroyed unit to be relevant

and btw, checking at #error-reports, I think there's a few more widgets affected, this is just the first one I have investigated.

**Comment by saurtron** [comment]
should already by fixed by https://github.com/beyond-all-reason/Beyond-All-Reason/pull/4100
